letsencrypt/client/tests/crypto_util_test.py

```diff
@@ -133,5 +134,71 @@ def test_san(self):
         self._call('cert-san.pem')
 
 
+class GetSansFromCsrTest(unittest.TestCase):
+    """Tests for letsencrypt.client.crypto_util.get_sans_from_csr."""
+    def test_extract_one_san(self):
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        csr = pkg_resources.resource_string(
+            __name__, os.path.join('testdata', 'csr.pem'))
+        result = get_sans_from_csr(csr)
+        self.assertEqual(result, ['example.com'])
+
+    def test_extract_two_sans(self):
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        csr = pkg_resources.resource_string(
+            __name__, os.path.join('testdata', 'csr-san.pem'))
+        result = get_sans_from_csr(csr)
+        self.assertEqual(result, ['example.com', 'www.example.com'])
+
+    def test_extract_six_sans(self):
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        csr = pkg_resources.resource_string(
+            __name__, os.path.join('testdata', 'csr-6sans.pem'))
+        result = get_sans_from_csr(csr)
+        self.assertEqual(
+            result, ["example.com", "example.org", "example.net",
+                     "example.info", "subdomain.example.com",
+                     "other.subdomain.example.com"])
+
+    def test_parse_non_csr(self):
+        from M2Crypto.X509 import X509Error
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        self.assertRaises(X509Error, get_sans_from_csr, "hello there")
+
+    def test_parse_no_sans(self):
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        csr = pkg_resources.resource_string(
+            __name__, os.path.join('testdata', 'csr-nosans.pem'))
+        self.assertRaises(ValueError, get_sans_from_csr, csr)
+
+    @mock.patch("M2Crypto.X509.load_request_string")
+    def test_parse_weird_m2crypto_output(self, mock_lrs):
+        # It's not clear how to reach this exception with invalid input,
+        # because M2Crypto is likely to raise X509Error rather than
+        # returning invalid output, but we can test the possibility with
+        # mock.
+        mock_lrs.as_text.return_value = "Something other than OpenSSL output"
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        self.assertRaises(ValueError, get_sans_from_csr, "input")
+
+class MakeCSRTest(unittest.TestCase):  # pylint: disable=too-few-public-methods
+    """Tests for letsencrypt.client.crypto_util.make_csr."""
+    def test_make_csr(self):
+        from letsencrypt.client.crypto_util import make_csr, get_sans_from_csr
```

Comment:
pls don't comma-import, rather split into two lines to keep more consistent with the rest of the code